### DISCUSSION TITLE
Is q a column vector?

### DISCUSSION THREAD

**Discussion by ajeecai** [qpsolvers/qpsolvers#36] [Q&A] [answered]
![image](https://user-images.githubusercontent.com/3023271/101377676-15424000-38ed-11eb-8146-76ddef8da241.png)
Usually x is  column vector, q^T x should requires q is column vector? (same question for h) 
But from the [example ](https://pypi.org/project/qpsolvers/)it looks not. 

Thanks

**Comment by stephane-caron** [comment, marked as answer]
`q` is indeed a column vector, that is, its shape is `(n,)` for some integer `n`. In the example, we make sure it is the case by reshaping it:
```python
q = dot(array([3., 2., 3.]), M).reshape((3,))
```
After this, `q.shape == (3,)`, i.e. `q` is a column vector of length 3.

**Comment by ajeecai** [comment]
Hi @stephane-caron , I am not a seasoned Pythoner, sorry if my question is naive :) 

What is different between reshape((3,)) and reshape((3,1)), to me it looks like the latter producing a column vector …

![image](https://user-images.githubusercontent.com/3023271/103739909-49666900-5031-11eb-830c-794094d18806.png)

**Comment by stephane-caron** [comment]
No worries, GIYF here and to learn the ropes of NumPy in general: check out for instance [this answer on StackOverflow](https://stats.stackexchange.com/a/285005).
